libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2515,8 +2515,69 @@ void TypeChecker::requireLValue(Expression const& _expression)
 	_expression.annotation().lValueRequested = true;
 	_expression.accept(*this);
 
+	if (_expression.annotation().isLValue)
+		return;
+
 	if (_expression.annotation().isConstant)
```

Comment:
Maybe it is better to extract all of this into a function called `findReasonForConstantAssignment -> string` or something like that, so it is clear that from here downwards we will always generate an error.